.eslintrc.js

```diff
@@ -0,0 +1,210 @@
+require('@rushstack/eslint-patch/modern-module-resolution');
+
+module.exports = {
+  env: {
+    browser: true,
+    es2022: true,
+  },
+  extends: ['plugin:react/recommended', 'airbnb', 'eslint:recommended'],
+  parserOptions: {
+    ecmaFeatures: {
+      jsx: true,
+    },
+    ecmaVersion: 2022,
+    sourceType: 'module',
+  },
+  plugins: ['react', 'react-redux', 'react-hooks', 'sonarjs', 'react-func'],
+  ignorePatterns: ['cypress/*'],
+  overrides: [
+    {
+      files: [
+        '*.spec.js',
+        '*.spec.jsx',
+        '*.test.js',
+        '*.test.jsx',
+        '*.spec.ts',
+        '*.spec.tsx',
+        '*.test.ts',
+        '*.test.tsx',
+      ],
+      rules: {
+        'no-magic-numbers': ['off'],
+        'max-len': ['off'],
+        'max-lines': ['off'],
+      },
+    },
+  ],
+  rules: {
+    indent: ['error', 2],
+    'linebreak-style': 0,
+    quotes: ['error', 'single'],
+    semi: ['error', 'always'],
+    'class-methods-use-this': ['off'],
+    "no-plusplus": ["off"],
+    'no-underscore-dangle': [
+      'error',
+      {
+        allow: ['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'],
+      },
+    ],
+    'no-magic-numbers': [
+      'error',
+      {
+        ignore: [0, 1, 2, 100],
+        ignoreArrayIndexes: true,
+        enforceConst: true,
+        detectObjects: false,
+      },
+    ],
+    'react/button-has-type': ['off'],
+    'no-console': ['off'],
+    'no-param-reassign': ['off'],
+    'consistent-return': ['off'],
+    'no-undef': ['off'],
+    'max-len': [
+      'error',
+      {
+        code: 90,
+        ignoreComments: true,
+        ignoreUrls: true,
+        ignoreRegExpLiterals: true,
+      },
+    ],
+    'max-params': ['error', 4],
+    'max-lines': ['error', 250],
+    'max-lines-per-function': ['off'],
+    'react-func/max-lines-per-function': ['error', 50],
+    complexity: ['error', 15],
+    'object-curly-newline': ['off'],
+    'import/no-extraneous-dependencies': ['off'],
+    'import/prefer-default-export': ['off'],
+    'sonarjs/cognitive-complexity': ['error', 15],
+    'sonarjs/no-collapsible-if': ['error'],
+    'sonarjs/no-collection-size-mischeck': ['error'],
+    'sonarjs/no-duplicate-string': ['error'],
+    'sonarjs/no-duplicated-branches': ['error'],
+    'sonarjs/no-extra-arguments': ['error'],
+    'sonarjs/no-identical-conditions': ['error'],
+    'sonarjs/no-identical-expressions': ['error'],
+    'sonarjs/no-identical-functions': ['error'],
+    'sonarjs/no-inverted-boolean-check': ['error'],
+    'sonarjs/no-one-iteration-loop': ['error'],
+    'sonarjs/no-redundant-boolean': ['error'],
+    'sonarjs/no-unused-collection': ['error'],
+    'sonarjs/no-use-of-empty-return-value': ['error'],
+    'sonarjs/no-useless-catch': ['error'],
+    'sonarjs/prefer-object-literal': ['error'],
+    'sonarjs/prefer-single-boolean-return': ['error'],
+    'react/jsx-filename-extension': [
+      1,
+      {
+        extensions: ['.js', '.jsx'],
```

Comment:
Aqui não deveria ser assim:

```suggestion
        extensions: ['.ts', '.tsx'],
```

ou eu to viajando?

Reply:
Essa config é do eslint sem typescript, a config do typescript está no arquivo typescript.js